crates/air_r_formatter/src/r/auxiliary/call_arguments_table.rs

```diff
@@ -0,0 +1,508 @@
+    // Take a snapshot of the formatter buffer to restore it on exit. We're
+    // eagerly formatting table cells in the buffer and need to undo that work.
+    let snapshot = f.snapshot();
+
+    let table = build_table_impl(args, f);
+
+    f.restore_snapshot(snapshot);
+    table
+}
+
+// First pass: Parse arguments into rows and gather column width information for
+// alignment. This involves finding the integer and fractional widths of numeric
+// arguments, and formatting other arguments in a flat layout to get the width of
+// the final printed text.
+fn build_table_impl(args: &RArgumentList, f: &mut RFormatter) -> FormatResult<Option<TableInfo>> {
+    let mut cols: Vec<ColumnInfo> = Vec::new();
+    let mut rows: Vec<Vec<ArgData>> = Vec::new();
+    let mut current_row = Vec::new();
+
+    let mut items = args.elements().enumerate();
+    let mut remaining = vec![];
+
+    for (i, arg) in &mut items {
+        // We've encountered an `off` comment before, keep collecting remaining args
+        if remaining.len() > 0 {
+            remaining.push(arg);
+            continue;
+        }
+
+        let arg_node = arg.node()?;
+        let arg_separator = arg.trailing_separator()?;
+
+        // If we see a named argument, start collecting remaining args. These
+        // will be formatted in fully expanded layout. The main idea is that
+        // table formatting is for unnamed arguments, and we allow trailing
+        // named arguments to parameterise the table function. See for instance
+        // `data.table::fcase(default = )` argument.
+        if arg_node.name_clause().is_some() {
+            remaining.push(arg);
+            continue;
+        }
+
+        // Detect leading line breaks because they indicate a new row
+        let lines_before = if arg_node.value().is_some() {
+            get_lines_before(arg_node.syntax())
+        } else {
+            arg_separator.map_or(0, get_lines_before_token)
+        };
+
+        // Push new row if any. Empty lines are not rows. Note empty lines still
+        // get formatted as part of trivia in the formatting pass. We just don't
+        // want to consider empty lines as rows in our table info.
+        if lines_before > 0 && !current_row.is_empty() {
+            rows.push(current_row);
+            current_row = Vec::new();
+        }
+
+        let column_index = current_row.len();
+
+        // Adjust number of columns in case this row grew longer than the
+        // current number of columns recorded. Note the table is allowed to be
+        // ragged so extending the columns might happen at every row. The loop
+        // should be unnecessary as we grow row elements one by one, it is just
+        // to be safe.
+        while cols.len() <= column_index {
+            cols.push(ColumnInfo::default());
+        }
+
+        let kind = match ArgKind::parse(arg_node, f)? {
+            Some(kind) => kind,
+            None => return Ok(None),
+        };
```

Comment:
```suggestion
        let Some(kind) = ArgKind::parse(arg_node, f)? else {
            // Argument has a forced newline, bail on table formatting
            return Ok(None);
        };
```

Had to go look up what it means to hit the `None` case, so it felt like it could use a comment